server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java

```diff
@@ -549,6 +549,65 @@ void capabilityShouldAdvertiseLoginOnUnEncryptedChannel() throws Exception {
         }
     }
 
+    @Nested
+    class AuthenticationRequireSSL {
+        IMAPServer imapServer;
+
+        @AfterEach
+        void tearDown() {
+            if (imapServer!=null){
+                imapServer.destroy();
+            }
+        }
+        @Test
+        void loginShouldFailWhenRequireSSLAndPlainAuthNotEnabled() throws Exception {
+            imapServer = createImapServer("imapServerPlainAuthDisabledAndRequireSSL.xml");
+            int port = imapServer.getListenAddresses().get(0).getPort();
+
+            assertThatThrownBy(() ->
+                testIMAPClient.connect("127.0.0.1", port)
+                    .login(USER.asString(), USER_PASS))
+                .hasMessage("Login failed");
+
+        }
+
+        @Test
+        void loginShouldSuccessWhenRequireSSLAndPlainAuthEnabled() throws Exception {
+            imapServer = createImapServer("imapServerPlainAuthEnabledAndRequireSSL.xml");
+            int port = imapServer.getListenAddresses().get(0).getPort();
+
+            assertThatCode(() ->
+                testIMAPClient.connect("127.0.0.1", port)
+                    .login(USER.asString(), USER_PASS)
+                    .append("INBOX", SMALL_MESSAGE))
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        void loginShouldSuccessWhenNOTRequireSSLAndPlainAuthNotEnabled() throws Exception {
```

Comment:
it not pass

Reply:
When Not required SSL and Plain Auth not enabled => I think login should fail